sdk/helper/random/string_generator.go

```diff
@@ -0,0 +1,311 @@
+package random
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"io"
+	"math"
+	"sort"
+	"time"
+	"unicode"
+
+	"github.com/hashicorp/go-multierror"
+)
+
+var (
+	LowercaseCharset   = sortCharset("abcdefghijklmnopqrstuvwxyz")
+	UppercaseCharset   = sortCharset("ABCDEFGHIJKLMNOPQRSTUVWXYZ")
+	NumericCharset     = sortCharset("0123456789")
+	FullSymbolCharset  = sortCharset("!\"#$%&'()*+,-./:;<=>?@[\\]^_`{|}~")
+	ShortSymbolCharset = sortCharset("-")
+
+	AlphabeticCharset              = sortCharset(UppercaseCharset + LowercaseCharset)
+	AlphaNumericCharset            = sortCharset(AlphabeticCharset + NumericCharset)
+	AlphaNumericShortSymbolCharset = sortCharset(AlphaNumericCharset + ShortSymbolCharset)
+	AlphaNumericFullSymbolCharset  = sortCharset(AlphaNumericCharset + FullSymbolCharset)
+
+	LowercaseRuneset   = []rune(LowercaseCharset)
+	UppercaseRuneset   = []rune(UppercaseCharset)
+	NumericRuneset     = []rune(NumericCharset)
+	FullSymbolRuneset  = []rune(FullSymbolCharset)
+	ShortSymbolRuneset = []rune(ShortSymbolCharset)
+
+	AlphabeticRuneset              = []rune(AlphabeticCharset)
+	AlphaNumericRuneset            = []rune(AlphaNumericCharset)
+	AlphaNumericShortSymbolRuneset = []rune(AlphaNumericShortSymbolCharset)
+	AlphaNumericFullSymbolRuneset  = []rune(AlphaNumericFullSymbolCharset)
+
+	// DefaultStringGenerator has reasonable default rules for generating strings
+	DefaultStringGenerator = StringGenerator{
+		Length: 20,
+		Rules: []Rule{
+			Charset{
+				Charset:  LowercaseRuneset,
+				MinChars: 1,
+			},
+			Charset{
+				Charset:  UppercaseRuneset,
+				MinChars: 1,
+			},
+			Charset{
+				Charset:  NumericRuneset,
+				MinChars: 1,
+			},
+			Charset{
+				Charset:  ShortSymbolRuneset,
+				MinChars: 1,
+			},
+		},
+	}
+)
+
+func sortCharset(chars string) string {
+	r := runes(chars)
+	sort.Sort(r)
+	return string(r)
+}
+
+// Rule to assert on string values.
+type Rule interface {
```

Comment:
Feels like this struct might make more sense in rules.go

Reply:
:+1: